service/matching/task_writer.go

```diff
@@ -177,6 +177,7 @@ func (w *taskWriter) appendTask(
 		metrics.TaskWriteThrottlePerTaskQueueCounter.With(w.tlMgr.metricsHandler).Record(1)
 		return nil, serviceerror.NewResourceExhausted(
 			enumspb.RESOURCE_EXHAUSTED_CAUSE_SYSTEM_OVERLOADED,
+			enumspb.RESOURCE_SCOPE_SYSTEM,
```

Comment:
Should this be SYSTEM or NAMESPACE? Task-queues are per namespace, so it filling up is a sign that a namespace is creating more tasks than they have pollers to process them

Reply:
Yeah that makes sense. It should be NAMESPACE.